`make deploy` fails with `AttributeError`

```
$ make deploy
make -C lambdas
...
Traceback (most recent call last):
  File "/home/nadove/PycharmProjects/azul-sc/lambdas/indexer/.chalice/config.json.template.py", line 59, in <module>
    indexer.forward_alb_logs.name: chalice.vpc_lambda_config(app_name),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'function' object has no attribute 'name'
make[2]: *** [/home/nadove/PycharmProjects/azul-sc/lambdas/../common.mk:119: .chalice/config.json] Error 1
make[2]: Leaving directory '/home/nadove/PycharmProjects/azul-sc/lambdas/indexer'
make[1]: *** [Makefile:12: indexer] Error 2
make[1]: Leaving directory '/home/nadove/PycharmProjects/azul-sc/lambdas'
make: *** [Makefile:98: lambdas] Error 2
```

--- a/lambdas/indexer/.chalice/config.json.template.py
+++ b/lambdas/indexer/.chalice/config.json.template.py
@@ -56,8 +56,14 @@ emit({
                     'lambda_timeout': config.aggregation_lambda_timeout(retry=True),
                     **chalice.vpc_lambda_config(app_name)
                 },
-                indexer.forward_alb_logs.name: chalice.vpc_lambda_config(app_name),
-                indexer.forward_s3_logs.name: chalice.vpc_lambda_config(app_name),
+                **(
+                    {
+                        indexer.forward_alb_logs.name: chalice.vpc_lambda_config(app_name),
+                        indexer.forward_s3_logs.name: chalice.vpc_lambda_config(app_name),
+                    }
+                    if config.enable_log_forwarding else
+                    {}
+                ),
                 **(
                     {
                         indexer.mirror.name: {

--- a/src/azul/service/repository_controller.py
+++ b/src/azul/service/repository_controller.py
@@ -265,7 +265,8 @@ class RepositoryController(SourceController):
 
         plugin = self.repository_plugin(catalog)
 
-        is_mirrored = self.mirror_service.is_mirrored(catalog, file)
+        is_mirrored = (config.enable_mirroring
+                       and self.mirror_service.is_mirrored(catalog, file))
         if is_mirrored:
             download = MirrorFileDownload(
                 file=file,

--- a/test/indexer/test_mirror_controller.py
+++ b/test/indexer/test_mirror_controller.py
@@ -2,16 +2,12 @@ import hashlib
 import json
 from unittest.mock import (
     MagicMock,
-    PropertyMock,
     patch,
 )
 
 import jsonschema
 from more_itertools import (
     one,
-)
-from moto import (
-    mock_aws,
 )
 
 from app_test_case import (
@@ -20,9 +16,6 @@ from app_test_case import (
 from azul import (
     R,
     config,
-)
-from azul.deployment import (
-    aws,
 )
 from azul.http import (
     http_client,
@@ -47,7 +40,7 @@ from azul_test_case import (
     DCP2TestCase,
 )
 from service import (
-    S3TestCase,
+    MirrorTestCase,
 )
 from sqs_test_case import (
     WorkQueueTestCase,
@@ -61,35 +54,18 @@ def setUpModule():
     configure_test_logging(log)
 
 
-@mock_aws
 class TestMirrorController(DCP2TestCase,
                            LocalAppTestCase,
                            WorkQueueTestCase,
-                           S3TestCase):
+                           MirrorTestCase):
 
     @classmethod
     def lambda_name(cls) -> str:
         return 'indexer'
 
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        cls.addClassPatch(patch.object(type(config),
-                                       'enable_mirroring',
-                                       new=PropertyMock(return_value=True)))
-        cls.addClassPatch(patch.object(type(config),
-                                       'mirror_bucket',
-                                       new=PropertyMock(return_value=None)))
-
-    @property
-    def bucket(self) -> str:
-        return aws.mirror_bucket
-
     def test_mirroring(self):
         self._create_mock_queues(config.mirror_queue.name,
                                  config.mirror_queue.to_fail.name)
-        self._create_test_bucket(self.bucket)
-
         with self.subTest('remote_mirror'):
             source_message = self._test_remote_mirror()
 
@@ -156,7 +132,7 @@ class TestMirrorController(DCP2TestCase,
         event = [self._mock_sqs_record(file_message)]
         with patch.object(MirrorService, '_download', return_value=self._file_contents):
             self.mirror_controller.mirror(event)
-        response = self._s3.get_object(Bucket=self.bucket,
+        response = self._s3.get_object(Bucket=self.mirror_bucket,
                                        Key=self.mirror_controller.service.mirror_object_key(file))
         mirrored_file_contents = response['Body'].read()
         self.assertEqual(mirrored_file_contents, self._file_contents)

--- a/test/indexer/test_mirror_controller.py
+++ b/test/indexer/test_mirror_controller.py
@@ -62,7 +62,10 @@ def setUpModule():
 
 
 @mock_aws
-class TestMirrorController(DCP2TestCase, LocalAppTestCase, WorkQueueTestCase, S3TestCase):
+class TestMirrorController(DCP2TestCase,
+                           LocalAppTestCase,
+                           WorkQueueTestCase,
+                           S3TestCase):
 
     @classmethod
     def lambda_name(cls) -> str:
@@ -151,7 +154,7 @@ class TestMirrorController(DCP2TestCase, LocalAppTestCase, WorkQueueTestCase, S3
 
     def _test_mirror_file(self, file, file_message):
         event = [self._mock_sqs_record(file_message)]
-        with patch.object(MirrorService, '_download', return_value=(self._file_contents)):
+        with patch.object(MirrorService, '_download', return_value=self._file_contents):
             self.mirror_controller.mirror(event)
         response = self._s3.get_object(Bucket=self.bucket,
                                        Key=self.mirror_controller.service.mirror_object_key(file))

--- a/test/service/__init__.py
+++ b/test/service/__init__.py
@@ -17,6 +17,7 @@ from typing import (
 )
 from unittest.mock import (
     MagicMock,
+    PropertyMock,
     patch,
 )
 import uuid
@@ -236,6 +237,20 @@ class StorageServiceTestCase(S3TestCase):
     def setUp(self) -> None:
         super().setUp()
         self._create_test_bucket(self.storage_service.bucket_name)
+
+
+class MirrorTestCase(S3TestCase):
+    mirror_bucket = 'test-mirror-bucket'
+
+    def setUp(self):
+        super().setUp()
+        self.addPatch(patch.object(type(config),
+                                   'enable_mirroring',
+                                   new=PropertyMock(return_value=True)))
+        self.addPatch(patch.object(type(config),
+                                   'mirror_bucket',
+                                   new=PropertyMock(return_value=self.mirror_bucket)))
+        self._create_test_bucket(self.mirror_bucket)
 
 
 @deprecated('Instead of decorating your test case, or its test methods in it, '

--- a/test/service/test_repository_files.py
+++ b/test/service/test_repository_files.py
@@ -14,6 +14,7 @@ from unittest import (
 )
 from unittest.mock import (
     MagicMock,
+    PropertyMock,
 )
 
 import attr
@@ -417,6 +418,9 @@ class TestRepositoryFilesWithMirroring(DCP2TestCase,
         cls.addClassPatch(mock.patch.object(BaseMirrorService,
                                             '_bucket_name',
                                             return_value=cls.bucket_name))
+        cls.addClassPatch(mock.patch.object(type(config),
+                                            'enable_mirroring',
+                                            new=PropertyMock(return_value=True)))
 
     def test_repository_files(self):
         file_content = b'Contents of foo'

--- a/test/service/test_repository_files.py
+++ b/test/service/test_repository_files.py
@@ -14,7 +14,6 @@ from unittest import (
 )
 from unittest.mock import (
     MagicMock,
-    PropertyMock,
 )
 
 import attr
@@ -27,9 +26,6 @@ from furl import (
 )
 from google.auth.transport.urllib3 import (
     AuthorizedHttp,
-)
-from moto import (
-    mock_aws,
 )
 import requests
 import responses
@@ -83,6 +79,7 @@ from azul_test_case import (
     DCP2TestCase,
 )
 from service import (
+    MirrorTestCase,
     S3TestCase,
 )
 
@@ -406,21 +403,9 @@ class TestRepositoryFilesWithDSS(DCP1TestCase,
                                 self.assertUrlEqual(re_pre_signed_s3_url, location)
 
 
-@mock_aws
 class TestRepositoryFilesWithMirroring(DCP2TestCase,
                                        RepositoryFilesTestCase,
-                                       S3TestCase):
-    bucket_name = 'test-bucket'
-
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        cls.addClassPatch(mock.patch.object(BaseMirrorService,
-                                            '_bucket_name',
-                                            return_value=cls.bucket_name))
-        cls.addClassPatch(mock.patch.object(type(config),
-                                            'enable_mirroring',
-                                            new=PropertyMock(return_value=True)))
+                                       MirrorTestCase):
 
     def test_repository_files(self):
         file_content = b'Contents of foo'
@@ -437,7 +422,6 @@ class TestRepositoryFilesWithMirroring(DCP2TestCase,
                        crc32c=None)
 
         mirror_service = MirrorService(schema_url_func=MagicMock())
-        self._create_test_bucket(self.bucket_name)
         with mock.patch.object(MirrorService, '_download', return_value=file_content):
             mirror_service.mirror_file(self.catalog, file)
         self.assertTrue(mirror_service.is_mirrored(self.catalog, file))
@@ -451,7 +435,7 @@ class TestRepositoryFilesWithMirroring(DCP2TestCase,
 
         signed_url = furl(response.headers['Location'])
         self.assertEqual('https', signed_url.scheme)
-        self.assertEqual(f'{self.bucket_name}.s3.{self._aws_test_region}.amazonaws.com',
+        self.assertEqual(f'{self.mirror_bucket}.s3.{config.region}.amazonaws.com',
                          signed_url.netloc)
         self.assertEqual('/' + mirror_service.mirror_object_key(file),
                          str(signed_url.path))

